File: android/DFM/src/main/java/com/dontflymoney/listeners/DialogNature.java
package com.dontflymoney.listeners;

import com.dontflymoney.view.MovesCreateActivity;
import com.dontflymoney.view.R;
import com.dontflymoney.viewhelper.DialogSelectClickListener;

import org.json.JSONArray;
import org.json.JSONException;

public class DialogNature extends DialogSelectClickListener
{
	private MovesCreateActivity activity;

	public DialogNature(JSONArray list, MovesCreateActivity activity)
	{
		super(list);
		this.activity = activity;
	}

	@Override
	public void setResult(String text, String value)
	{
		activity.setNature(text, value);
	}

	@Override
	public void handleError(JSONException exception)
	{
		activity.message.alertError(R.string.error_convert_result);
	}
}

File: android/DFM/src/main/kotlin/com/darakeon/uiHelpers/listeners/DialogNature.kt
package com.dontflymoney.listeners

import com.dontflymoney.view.MovesCreateActivity
import com.dontflymoney.view.R
import com.dontflymoney.viewhelper.DialogSelectClickListener

import org.json.JSONArray
import org.json.JSONException

class DialogNature(list: JSONArray?, private val activity: MovesCreateActivity) : DialogSelectClickListener(list) {

    override fun setResult(text: String, value: String) {
        activity.setNature(text, value)
    }

    override fun handleError(exception: JSONException) {
        activity.message.alertError(R.string.error_convert_result)
    }
}
